recipe/meta.yaml

```diff
@@ -44,6 +42,7 @@ outputs:
     script: build-core.bat  # [win]
     requirements:
       build:
+        - {{ compiler('go1.4-bootstrap') }}
         {% if not win %}
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
```

Comment:
Are these still needed then?

Reply:
I am investigating this issue as well. The crux of it is that we can build `go` with or without `cgo` enabled. 

The default build is with CGO enabled, which means we need the compilers during build time as well as runtime if the underlying library needs to link against C libraries. I noticed more packages linking against C libraries than before, but my sample is biased because I tend to build infrastructure packages like Terraform, Mongo-tools, etc.

We can't cross-compile anything with CGO because we don't have proper cross-compilers, so it should be disabled when building the go_<platform> packages when the platform does not match the host_platform, e.g. `go_linux-64` built on `osx-64`.

It is a matter of choice now, and I would like to not have to make a decision on my own since these packages are also feeding defaults, and I rather keep it consistent with that they would accept.